Add my-wallet-assignment day-1 task

MoFahimKh opened the pull request:
- This code is an assignment of a react-app  with functional components that interacts with Ethereum blockchain and its testnet works.

- The app components is the main component which uses ConnectButton components , which connects us to the desired Ethereum chain and returns the wallet address as well as balance of eEhers in the wallet.....

- The next components is Interactions component which is called in the ConnectButton components and it interacts to send transactions to the desired ethereum address , takes two inputs i.e receiver's account's address and the amount of ethers that are to be sent to the receiver ..covering other edge cases and handeling error for the same . Thank you!

MFahimKhQodeleaf committed:
Add my-wallet-assignment day-1 task

abbasali1994 commented on src/connectButton.js:
```diff
@@ -0,0 +1,126 @@
+import { React, useEffect, useState } from "react";
```
Rename it as pages/Home.js

abbasali1994 commented on src/connectButton.js:
```diff
@@ -0,0 +1,126 @@
+    //   alert(err.data.err);
+    // }
+
+    const balance = await window.ethereum.request({
```
Rename it as utils/ethereum.js

getbalance function

abbasali1994 commented on src/connectButton.js:
```diff
@@ -0,0 +1,126 @@
+        if (await account[0]) {
+          setButtonText("Connected Wallet!");
+          setWalletAddress(
+            (account = await window.ethereum.request({
```
get account function in utils

abbasali1994 commented on src/interaction.js:
```diff
@@ -0,0 +1,61 @@
+    try {
+      const provider = new ethers.providers.Web3Provider(window.ethereum);
+      const signer = provider.getSigner();
+      const receiversAddress = document.getElementById("recAddress").value;
```
incorrect

MFahimKhQodeleaf committed:
Update my-wallet-assginment :
	1.) Add network logic
	2.) Integrate react bootstrap
	3.) Modularize src file
	4.) Add Polygon Mumbai testnet in the assignment

abbasali1994 commented on src/pages/Home.js:
```diff
@@ -0,0 +1,101 @@
+import Navbar from "react-bootstrap/Navbar";
+import NavDropdown from "react-bootstrap/NavDropdown";
+
+const ConnectButton = () => {
```
Rename Home

abbasali1994 commented on src/SendToken.js:
```diff
@@ -0,0 +1,51 @@
+
+  return (
+    <div className="formContainer">
+      <Form>
```
use Form components

MFahimKhQodeleaf committed:
Update my-wallet-app
	1.)Add components folder
	2.)send Token fixes, correcting the network logic
	3.)Provide USDT token transfer functionality
	4.)Reduce redundancy

MFahimKhQodeleaf committed:
fix network change issue in transactions.js utility file

abbasali1994 commented on src/components/Navbar.js:
```diff
@@ -0,0 +1,120 @@
+import Navbar from "react-bootstrap/Navbar";
+import NavDropdown from "react-bootstrap/NavDropdown";
+import Container from "react-bootstrap/Container";
+import { GetAccount, GetBalance } from "../utils/account";
```
GetAccount, GetBalance rename to camelcase

abbasali1994 commented on src/components/SendToken.js:
```diff
@@ -0,0 +1,74 @@
+import { Button } from "react-bootstrap";
+import { sendEther, sendUSDT } from "../utils/transactions";
+
+const SendTransactions = () => {
```
Still filename incorrect

abbasali1994 commented on src/utils/transactions.js:
```diff
@@ -0,0 +1,39 @@
+  try {
+    const provider = new ethers.providers.Web3Provider(window.ethereum);
+    const signer = provider.getSigner();
+    const USDTAddress = "0xdAC17F958D2ee523a2206206994597C13D831ec7";
```
This address will be dynamic as it will chnge for diffrent networks, make it such that address is picked as per set network

MFahimKhQodeleaf committed:
Update naming conventions

MFahimKhQodeleaf committed:
Update naming conventions

MFahimKhQodeleaf committed:
Update src:
	1.)Add context
	2.)Add custom hook
	3.)Add Weth erc20 token on sepolia
	4.)Add UI checks

MFahimKhQodeleaf committed:
Remove unwanted commented code

MFahimKhQodeleaf committed:
Remove unwanted console logs and fix indent

MoFahimKh committed:
Merge branch 'main' into my-wallet-app
